### PR TITLE
show progress indicator for implementation cmd

### PR DESCRIPTION
**Pull request by ramya-rao-a**
This PR adds the progress indicator for the `Go to implementation` command.
See #31666

cc @jrieken @mjbvz

### CODE REVIEW

**Review by jrieken — CHANGES_REQUESTED** [review]
There is something not good with the change to `monaco.d.ts`

**Comment by ramya-rao-a** [comment]
@jrieken Sorry abt that! Didn't notice that slipping through. Fixed

**Comment by ramya-rao-a** [comment]
@jrieken The monaco.d.ts file is removed from the PR. Rest looks good?

**Comment by jrieken** [comment]
Sorry, did hit reply yesterday. Yeah, the change looks good, consider moving this up to the base class, otherwise feel free to merge

File: src/vs/editor/contrib/goToDeclaration/browser/goToDeclarationCommands.ts [hunk 1]
```diff
@@ -279,6 +280,13 @@ export class GoToImplementationAction extends ImplementationAction {
 			}
 		});
 	}
+
+	public run(accessor: ServicesAccessor, editor: editorCommon.ICommonCodeEditor): TPromise<void> {
```
Comment (ramya-rao-a) [inline src/vs/editor/contrib/goToDeclaration/browser/goToDeclarationCommands.ts]: you mean moving it to `ImplementationAction` or `DefinitionAction` ?

Comment (mjbvz) [inline src/vs/editor/contrib/goToDeclaration/browser/goToDeclarationCommands.ts]: I'd say move it to `DefinitionAction`. That way, `go to definition` and `go to type definition` also pick it up

**Review by mjbvz — APPROVED** [review]

File: src/vs/editor/contrib/goToDeclaration/browser/goToDeclarationCommands.ts (hunk 1 above)
Comment (ramya-rao-a) [inline src/vs/editor/contrib/goToDeclaration/browser/goToDeclarationCommands.ts]: Done

**Comment by jrieken** [comment]
Thanks!

**Pull request merged**